tests/test_functors.py

```diff
@@ -42,119 +42,181 @@
 ROOT = os.path.abspath(os.path.dirname(__file__))
 
 
-def setup_module(module):
-    lsst.utils.tests.init()
-
-
 @unittest.skipUnless(havePyArrow, "Requires pyarrow")
 class FunctorTestCase(unittest.TestCase):
+
+    def simulateMultiParquet(self, dataDict):
+        """Create a simple test MultilevelParquetTable
+        """
+        simpleDF = pd.DataFrame(dataDict)
+        dfFilterDSCombos = []
+        for ds in self.datasets:
+            for filterName in self.filters:
+                df = copy.copy(simpleDF)
+                df.reindex(sorted(df.columns), axis=1)
+                df['dataset'] = ds
+                df['filter'] = filterName
+                df.columns = pd.MultiIndex.from_tuples(
+                    [(ds, filterName, c) for c in df.columns],
+                    names=('dataset', 'filter', 'column'))
+                dfFilterDSCombos.append(df)
```

Comment:
Relating to several comments below, generating this fake data should make sure that different filters/datasets have different values in their columns, not the same, otherwise the equality/inequality tests later on lose their meaning.

Reply:
I would argue that such a test should be performed on a more generic functor, making sure that different datasets/filters with different values return said different values. Testing these on functors with a specific use seems to conflate two tests 1) that the internal calculations return the values expected (not being currently tested except in the case that color = mag1 - mag2), 2) that fuctors properly access different dataset/filter columns (tested currently within specific functor unittests rather than a specific test case for the functionality.) Like I said, this won't sit idle if we don't address it in this ticket. I'll be working on the functors and the SDM for the next few months so I can make a ticket to specifically write a test to test the dataset/filter functionality.

Reply:
Yup, sounds good.  As you see, these tests were originally pretty bare-bones, so I do think it's a good idea to flesh them out to make sure they are computing the right things.